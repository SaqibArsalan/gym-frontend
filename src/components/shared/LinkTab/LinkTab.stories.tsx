// Generated with util/create-component.js
import React from 'react';
import LinkTab from './LinkTab';

export default {
	title: 'LinkTab',
};

export const WithIcon = () => (
	<LinkTab label='With icon'  href='#' />
);

export const WithoutIcon = () => <LinkTab label='No icon' href='#' />;
