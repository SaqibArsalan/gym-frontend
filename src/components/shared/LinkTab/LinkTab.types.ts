/* eslint-disable no-unused-vars */
import React from 'react';

// Generated with util/create-component.js
// eslint-disable-next-line no-shadow
export enum IconPositionEnum {
	top = 'top',
	start = 'start',
	end = 'end',
	bottom = 'bottom',
}

export interface ILinkTabProps {
	label: string;
	href: string;
	onClick?: (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => void;
	children?: any;
	icon?:
		| string
		| React.ReactElement<any, string | React.JSXElementConstructor<any>>;
	iconPosition?: IconPositionEnum;
	value?: number;
}
