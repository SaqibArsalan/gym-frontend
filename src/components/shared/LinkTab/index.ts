export { default as LinkTab } from './LinkTab';
export { IconPositionEnum } from './LinkTab.types';
