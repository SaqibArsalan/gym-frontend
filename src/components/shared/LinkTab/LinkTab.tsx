// Generated with util/create-component.js
import React from 'react';
import Tab from '@mui/material/Tab';

import { IconPositionEnum, ILinkTabProps } from './LinkTab.types';

import './LinkTab.scss';

const LinkTab: React.FC<ILinkTabProps> = (props: ILinkTabProps) => (
	<Tab {...props} />
);

LinkTab.defaultProps = {
	onClick: (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
		event.preventDefault();
	},
	iconPosition: IconPositionEnum.start,
};

export default LinkTab;
