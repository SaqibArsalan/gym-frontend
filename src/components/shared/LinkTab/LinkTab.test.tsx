// Generated with util/create-component.js
import React from 'react';
import { mount, ReactWrapper } from 'enzyme';

import LinkTab from './LinkTab';
import { ILinkTabProps } from './LinkTab.types';

describe('Link Tab Component', () => {
	const props: ILinkTabProps = {
		href: '#',
		label: 'link tab',
	};
	let wrapper: ReactWrapper<
		any,
		Readonly<{}>,
		React.Component<ILinkTabProps, {}, any>
	>;

	beforeEach(() => {
		wrapper = mount(<LinkTab {...props} />);
	});

	test('component should successfully render', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('should contain tab text', () => {
		const tabText = wrapper.find('a').first();
		// console.log('tabText', tabText.text);
		expect(tabText.text()).toEqual(props.label);
	});
});
