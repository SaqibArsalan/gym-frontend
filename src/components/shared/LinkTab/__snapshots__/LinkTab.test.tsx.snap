// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Tab Component component should successfully render 1`] = `ReactWrapper {}`;
